### PR TITLE
convert-meta-tags: don't allow newlines when converting meta tags.

### PR DESCRIPTION
**Pull request by oschaaf**
This change makes ResponseHeaders::MergeContentType reject values
containing unprintable characters.

Fixes https://github.com/pagespeed/mod_pagespeed/issues/1083

### CODE REVIEW

File: pagespeed/kernel/http/response_headers.cc [hunk 1]
```diff
@@ -321,23 +321,28 @@ bool ResponseHeaders::CombineContentTypes(const StringPiece& orig,
...
-  if (old_values.size() < 1) {
-    ret = CombineContentTypes("", content_type);
-  } else if (old_values.size() == 1) {
-    StringPiece old_val(*old_values[0]);
-    ret = CombineContentTypes(old_val, content_type);
-  }
+
   if (ret) {
-    cache_fields_dirty_ = true;
+    Lookup(HttpAttributes::kContentType, &old_values);
```
Comment (crowell) [inline pagespeed/kernel/http/response_headers.cc]: check return value of `Lookup`, false means lookup failed.

Comment (oschaaf) [inline pagespeed/kernel/http/response_headers.cc]: @crowell  This is exactly like it was before this change. Double checking, it looks to me like this should work ok when `Lookup` would fail (`old_values` should be unchanged). Do you have additional handling in mind for the case where `Lookup` returns false?

Comment (crowell) [inline pagespeed/kernel/http/response_headers.cc]: I think i was just looking at the diff wrong in github, your change is unrelated to this block, sorry for the noise.

File: pagespeed/kernel/http/response_headers.cc [hunk 2]
```diff
@@ -321,23 +321,28 @@ bool ResponseHeaders::CombineContentTypes(const StringPiece& orig,
 }
 
 bool ResponseHeaders::MergeContentType(const StringPiece& content_type) {
-  bool ret = false;
+  bool ret = std::find_if(
+      content_type.begin(), content_type.end(), iscntrl) == content_type.end();
```
Comment (morlovich) [inline pagespeed/kernel/http/response_headers.cc]: I have to admit to a certain paranoia about the likes of iscntrl, due to their locale-sensitivity.
Maybe our inline bool IsNonControlAscii(char c) from string_util.h would be an acceptable substitute?

Comment (oschaaf) [inline pagespeed/kernel/http/response_headers.cc]: @morlovich thanks, sounds like a good idea, I'll swap in `IsNonControlAscii()`

Comment (oschaaf) [inline pagespeed/kernel/http/response_headers.cc]: @morlovich updated & force pushed

**Comment by jeffkaufman** [comment]
LGTM

I'll merge this.

**Comment by jeffkaufman** [comment]
Merged as 2cea12ce1157ff4e7ae0944578eeeef88110eacf
